src/Chapter4.hs

```diff
@@ -487,8 +502,33 @@ Implement the 'Applicative' instance for our 'List' type.
   apply each function to each argument and combine all the results. You
   may also need to implement a few useful helper functions for our List
   type.
+
+>>> Empty <*> Empty
+Empty
+
+>>> (Cons (+1) (Cons (+2) Empty)) <*> (Cons 1 (Cons 2 (Cons 3 Empty)))
+Cons 2 (Cons 3 (Cons 4 (Cons 3 (Cons 4 (Cons 5 Empty)))))
 -}
 
+listAppend :: List a -> List a -> List a
+listAppend Empty ys = ys
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
listAppend xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤